src/util/locktime.rs

```diff
@@ -0,0 +1,516 @@
+/// whether `LockTime < LOCKTIME_THRESHOLD`.
+///
+/// ### Relevant BIPs
+///
+/// * [BIP-65 OP_CHECKLOCKTIMEVERIFY](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)
+/// * [BIP-113 Median time-past as endpoint for lock-time calculations](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki)
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub enum LockTime {
+    /// A block height lock time value.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitcoin::locktime::{self, LockTime};
+    ///
+    /// let n_lock_time: u32 = 741521;
+    /// assert!(locktime::is_block_height(n_lock_time));
+    /// let lock_time = LockTime::from(n_lock_time);
+    ///
+    /// assert!(lock_time.is_block_height());
+    /// assert_eq!(lock_time.to_u32(), n_lock_time);
+    ///
+    /// let n = LockTime::from(741520); // From `741520 OP_CHECKLOCKTIMEVERIFY`.
+    /// assert!(n.is_same_unit(lock_time));
+    /// assert!(n.is_satisfied(lock_time).expect("same time lock units don't error"));
+    /// ```
+    Blocks(Height),
+    /// A UNIX timestamp lock time value.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitcoin::locktime::{self, LockTime};
+    ///
+    /// let n_lock_time: u32 = 1653195600; // May 22nd, 5am UTC.
+    /// assert!(locktime::is_block_time(n_lock_time));
+    /// let lock_time = LockTime::from(n_lock_time);
+    //
+    /// assert!(lock_time.is_block_time());
+    /// assert_eq!(lock_time.to_u32(), n_lock_time);
+    ///
+    /// let n = LockTime::from(1653195000); // From `1653195000 OP_CHECKLOCKTIMEVERIFY`.
+    /// assert!(n.is_same_unit(lock_time));
+    /// assert!(n.is_satisfied(lock_time).expect("same time lock units don't error"));
+    /// ```
+    Seconds(Time),
+}
+
+impl LockTime {
+    /// Returns true if both lock times use the same unit i.e., both height based or both time based.
+    pub fn is_same_unit(&self, other: LockTime) -> bool {
+        mem::discriminant(self) == mem::discriminant(&other)
+    }
+
+    /// Returns true if this lock time value is a block height.
+    pub fn is_block_height(&self) -> bool {
+        match *self {
+            LockTime::Blocks(_) => true,
+            LockTime::Seconds(_) => false,
+        }
+    }
+
+    /// Returns true if this lock time value is a block time (UNIX timestamp).
+    pub fn is_block_time(&self) -> bool {
+        !self.is_block_height()
+    }
+
+    /// Returns true if the timelock constraint of this `n OP_CHECKLOCKTIMEVERIFY` and nLockTime is
+    /// satisfied. To be clear, `self` is `n` and `n_lock_time` is the transaction nLockTime.
+    ///
+    /// # Errors
+    ///
+    /// If `n` and `n_lock_time` are different units i.e., one represents a block height and the
+    /// other represents a block time.
+    pub fn is_satisfied(&self, n_lock_time: LockTime) -> Result<bool, Error> {
```

Comment:
Did you forget to change it to accept both time and height?

Reply:
oh yes, my bad, thank you.